--- /usr/lib/user-setup/user-setup-apply	2012-10-15 03:31:33.000000000 -0600
+++ user-setup-apply	2015-05-06 21:43:02.854506004 -0600
@@ -398,4 +398,15 @@
 	db_set passwd/user-password-again ''
 fi
 
+DISTRO=`cat /etc/lsb-release | grep DISTRIB_ID | cut -f 2 -d "=" | sed -e 's/"//g'`
+DISTRO_VERSION=`cat /etc/lsb-release | grep DISTRIB_RELEASE | cut -f 2 -d "=" | sed -e 's/"//g'`
+DISTRO_NAME="${DISTRO}_${DISTRO_VERSION}"
+MACHINE=`$ROOT/usr/sbin/dmidecode | grep Product | cut -f 2 -d ":" | sed -e 's/[^a-zA-Z0-9_]*//g'`
+MACHINE_DIR="/var/lib/distroshare-updater/updates/${MACHINE}_${DISTRO_NAME}/"
+if [ -d "${ROOT}/${MACHINE_DIR}/etc/skel" ]
+  then
+  $chroot $ROOT /usr/bin/rsync -a "${MACHINE_DIR}/etc/skel/" /home/$USER/
+  $chroot $ROOT chown -R "$USER:$USER" "/home/$USER" >/dev/null || true
+fi
+
 exit 0
